openfpga_flow/tasks/fpga_bitstream/generate_bitstream/configuration_chain/device_4x4_PathBitSetting/config/write_xml.openfpga

```diff
@@ -0,0 +1,79 @@
+# This file is copied from fix_device_example_script.openfpga
+
+# Run VPR for the 'and' design
+#--write_rr_graph example_rr_graph.xml
+vpr ${VPR_ARCH_FILE} ${VPR_TESTBENCH_BLIF} --clock_modeling route --device ${OPENFPGA_VPR_DEVICE_LAYOUT}
+
+# Read OpenFPGA architecture definition
+read_openfpga_arch -f ${OPENFPGA_ARCH_FILE}
+
+# Read OpenFPGA simulation settings
+read_openfpga_simulation_setting -f ${OPENFPGA_SIM_SETTING_FILE}
+
+# Read OpenFPGA bitstream settings
+read_openfpga_bitstream_setting -f ${OPENFPGA_BITSTREAM_SETTING_FILE}
+
+# Annotate the OpenFPGA architecture to VPR data base
+# to debug use --verbose options
+link_openfpga_arch --activity_file ${ACTIVITY_FILE} --sort_gsb_chan_node_in_edges
+
+# Check and correct any naming conflicts in the BLIF netlist
+check_netlist_naming_conflict --fix --report ./netlist_renaming.xml
+
+# Apply fix-up to Look-Up Table truth tables based on packing results
+lut_truth_table_fixup
+
+# Build the module graph
+#  - Enabled compression on routing architecture modules
+#  - Enable pin duplication on grid modules
+build_fabric --compress_routing #--verbose
+
+# Write the fabric hierarchy of module graph to a file
+# This is used by hierarchical PnR flows
+write_fabric_hierarchy --file ./fabric_hierarchy.txt
+
+# Repack the netlist to physical pbs
+# This must be done before bitstream generator and testbench generation
+# Strongly recommend it is done after all the fix-up have been applied
+repack #--verbose
+
+# Build the bitstream
+#  - Output the fabric-independent bitstream to a file
+build_architecture_bitstream --verbose --write_file fabric_independent_bitstream.xml
```

Comment:
Please place the testcase under ``fpga_bitstream/overwrite_bitstream``
Also, the current testcase cannot validate if the overwrite does work or not. 

Refer to the following testcase: 

https://github.com/lnis-uofu/OpenFPGA/blob/master/openfpga_flow/tasks/fpga_bitstream/load_external_architecture_bitstream/config/task.conf

My suggestion is to load a bitstream file where you manually flip the LUT bits. For example, force it to malfunction by changing AND logic to OR logic. Then use your overwrite function to correct it.

Reply:
I move test case to fpga_bitstream/overwrite_bitstream. It is done via https://github.com/chungshien/OpenFPGA/commit/933155b08fe9e6b3d2d5bb3133ffbbfb0b5822dd

I had found a way to run validation script by calling ext_exec command, which previously I have difficulty to find a way to run my validation script. The new flow would be:
1. First the test generate a golden-bitstream which it is without any overwrite
2. Then the test randomly determine 200 paths to overwrite (every time we run the test, new 200 random paths will be determined)
3. Then generate test-bitstream which we overwrite these 200 paths via bitstream_annotation.xml and their values are inverted of golden bitstream
4. Compare golden-bitstream and test-bitstream:
    - those 200 paths: their value need be inverted
    - the rest paths: their value must still be the same

Reply:
It should work then. Good catch